apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/server/TracingServerCall.java

```diff
@@ -0,0 +1,111 @@
+
+    private final String operationPrefix;
+    private final ContextSnapshot contextSnapshot;
+
+    protected TracingServerCall(ServerCall<REQUEST, RESPONSE> delegate, ContextSnapshot contextSnapshot) {
+        super(delegate);
+        this.operationPrefix = OperationNameFormatUtil.formatOperationName(delegate.getMethodDescriptor()) + SERVER;
+        this.contextSnapshot = contextSnapshot;
+    }
+
+    @Override
+    public void sendMessage(RESPONSE message) {
+        // We just create the request on message span for server stream calls.
+        if (!getMethodDescriptor().getType().serverSendsOneMessage()) {
+            try {
+                final AbstractSpan span = ContextManager.createLocalSpan(operationPrefix + RESPONSE_ON_MESSAGE_OPERATION_NAME);
+                span.setComponent(ComponentsDefine.GRPC);
+                span.setLayer(SpanLayer.RPC_FRAMEWORK);
+                ContextManager.continued(contextSnapshot);
+                super.sendMessage(message);
+            } catch (Throwable t) {
+                ContextManager.activeSpan().errorOccurred().log(t);
+                throw t;
+            } finally {
+                ContextManager.stopSpan();
+            }
+        } else {
+            super.sendMessage(message);
+        }
+    }
+
+    @Override
+    public void close(Status status, Metadata trailers) {
+        try {
+            final AbstractSpan span = ContextManager.createLocalSpan(operationPrefix + RESPONSE_ON_CLOSE_OPERATION_NAME);
+            span.setComponent(ComponentsDefine.GRPC);
+            span.setLayer(SpanLayer.RPC_FRAMEWORK);
+            ContextManager.continued(contextSnapshot);
+
+            switch (status.getCode()) {
+                case OK:
+                    break;
+                // UNKNOWN/INTERNAL status code will case error in this span.
+                // Those status code means some unexpected error occurred in server.
+                // Similar to 5XX in HTTP status.
+                case UNKNOWN:
+                case INTERNAL:
+                    if (status.getCause() == null) {
+                        span.errorOccurred().log(status.asRuntimeException());
+                    } else {
+                        span.errorOccurred().log(status.getCause());
+                    }
+                    break;
+                // Other status code means some predictable error occurred in server.
+                // Like PERMISSION_DENIED or UNAUTHENTICATED somethings.
+                // Similar to 4XX in HTTP status.
+                default:
+                    // But if the status still has cause exception, we will log it too.
+                    if (status.getCause() != null) {
+                        span.errorOccurred().log(status.getCause());
+                    }
+                    break;
+            }
+            Tags.STATUS_CODE.set(span, status.getCode().name());
+            super.close(status, trailers);
```

Comment:
You should add a new `try/final` in this code block, and keep `super#close` in the final.